proposals/core.md

```diff
@@ -64,40 +65,42 @@ Responses from the server include the "id" of the command they are responding to
+
+Some events that are crucial to target discovery will be enabled by default. These are discussed in the Target Discovery section below.
 
-*Subscribing to an event*
+The first step to receive events on the client side would be to send a "subscribe" command:
+
+*Subscribing to events*
 
 ```json
 {
     "id": 0,
+    "to": "window/222",
     "method": "subscribe",
-    "params": { "event": "scriptContextCreated" }
+    "params": { "domains": ["logging", "network"] }
 }
 ```
 
-Sending this command would tell the server to start firing an event (i.e. "scriptContextCreated") to the client. The client would send a matching "unsubscribe" command when they no longer want to receive that event.
-
-Subscriptions for each event should be ref-counted on the server side. Calling subscribe would increment the ref count for an event and calling unsubscribe would decrement the ref count. The first time the client calls subscribe, the ref count goes from 0 to 1, and the WebDriver implementation would perform whatever browser-specific steps are needed to begin generating the event. When the ref count falls back to 0, the WebDriver implementation would stop generating the event.
+The "subscribe" command is sent to a top-level Target, in this case a Window. Sending this command tells the server to start sending logging- and network- related events to the client. The scope of a "subscribe" command is a top-level Target. This will enable events on all browsing contexts and realms that belong to that target. Enabling events for individual browsing contexts or realms is not supported. To enable events for a different top-level Target, the client must send another "subscribe" command.
 
-Ref counting is useful here because it would allow multiple consumers on the client side to call "subscribe". For example, some test code might want to subscribe for an event, and the code might use some third-party helper library that also wants to subscribe for the event. Ref counting allows the test code and helper library to subscribe and unsubscribe independently. Otherwise, the first one to call "unsubscribe" would inadvertently shut down events for both consumers.
+The client sends a matching "unsubscribe" command when they no longer want to receive events from a Target.
 
 *Sample event*
 
-Event messages don't have an "id" property since they are fire-and-forget.
+All event messages have a "from" property indicating which object in the BiDi model sent the event. Event messages don't have an "id" property since they are not associated with a command request message.
 
 ```json
 {
-    "method": "scriptContextCreated",
-    "params": { "scriptContextId": "<ID>" }
+    "from": "browsingContext/333",
+    "method": "logging.entryAdded",
+    "params": { "type": "info", message: "Hello World!" }

```

Comment:
```suggestion
    "params": { "type": "info", "message": "Hello World!" }

```

The syntax highlighting is messed up after this, so I'll commit this suggestion myself.